### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by ethred**
We implement the following features in this branch :-
- Make the code followed the principle of YAGNI, KISS, and DRY rules and Ruby style guide are respected.
- Follow SOLID principles.
- Apply  the "Single-responsibility Principle".
- The role of main.rb is only to start  the code and ask the user for the initial choice of options (or to quit).
- Extract some pieces of logic into separate class(es).

### CODE REVIEW

File: classes/app.rb [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Name: '
+    name = gets.chomp.to_s
+
+    print 'Specialization: '
+    specialization = gets.chomp.to_s
+
+    teacher = Teacher.new(specialization, age, name)
```
Comment (zeff96) [inline classes/app.rb]: - [ ] 🎓👨‍🏫👩‍🏫 To follow the single responsibility principle effectively, the calls to create instances of the Teacher and Student classes, which both inherit from the Person class, should be consolidated within the create_person method. 🧐 This method has the exclusive responsibility of crafting either a teacher or a student object, making our code more organized and adherent to the principle. 🛠️ It's all about keeping things neat and efficient! 💡😊

File: classes/app.rb [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Name: '
+    name = gets.chomp.to_s
+
+    print 'Has parent permission? [Y / N]: '
+    parent_permission = gets.chomp.to_s
```
Comment (zeff96) [inline classes/app.rb]: - [ ] 🎓👨‍🏫👩‍🏫 To follow the single responsibility principle effectively, all requests for input and input validation should've been placed in the main.rb file and/or classes/methods whose responsibilities are to request or validate input. consider below as an example to help you refactor further

```sh
  private

  def prompt_input(message)
    print message
    gets.chomp
  end

  def prompt_yes_no(message)
    val = prompt_input("#{message} (Y/N): ").downcase
    %w[y yes].include?(val)
  end
```

**Review by zeff96 — CHANGES_REQUESTED** [review]
### Status: CHANGES REQUIRED ♻️ ♻️ ♻️ 
Hi @ethred and @ngalamac , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights

:heavy_check_mark: Linters passing
:heavy_check_mark: Correct GitFlow

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by shahadat3669 — APPROVED** [review]
Hi @ethred, @ngalamac, 

Your project is complete!  There is nothing else to say other than it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ Followed the single responsibility principle.
✅ Updated code based on previous reviewer comments.
✅ The code is top-notch! 💯

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._



Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table  width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadata3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**